CAIPs/caip-25.md

```diff
@@ -16,422 +16,247 @@ CAIP-25 defines an authorization procedure for a chain agnostic provider to inte
 
-When a user wishes to update the authorizations of an active session from within the wallet, the wallet should notify the caller of the changes with a [`wallet_sessionChanged`][CAIP-311] notification.
+Subsequent `wallet_createSession` calls may:
 
-If a connection is initially established without a `sessionId` and the wallet later implements `sessionId` support, the wallet can revoke the single session and notify the caller via `wallet_sessionChanged`. When the caller seeks to re-establish the session via `wallet_createSession`, the wallet should return a `sessionId` in the response.
+- Update an existing session by including the same `sessionId`
+- Create a new session if no `sessionId` is provided (discouraged)
 
-When a caller wishes revoke an active session, it can do so by calling [`wallet_revokeSession`][CAIP-285].
-- When a `sessionId` is returned in the initial `wallet_createSession` response, the caller MUST call `wallet_revokeSession` with the supplied `sessionId` to revoke that session.
-- When the wallet does not provide a `sessionId` in its initial response, a call to `wallet_revokeSession` revokes the single active session between caller and wallet.
+Session updates triggered by the wallet must notify the caller using `wallet_sessionChanged`.
 
-For more detail on the lifecycle and management of sessions with and without `sessionId`s, see the informational [CAIP-316][].
+Callers may revoke sessions using `wallet_revokeSession`, either by `sessionId` or directly if no sessionId was established.
 
 #### Session Data and Metadata
 
-Initial and ongoing authorization requests are grouped into two top-level objects containing keyed [scopeObjects][CAIP-217], named `requiredScopes` and `optionalScopes`
-respectively.
-Each `scopeObject` in either parent object MUST be keyed uniquely within its parent, but these keys CAN appear in both
-(i.e., additional properties of an authorization target in `requiredScopes` may be requested in a separate `scopeObject` with the same key in the `optionalScopes` array).
+Authorization requests are expressed as a top-level object `scopes` containing keyed [scopeObjects][CAIP-217].
 
-Each `scopeObject` in these parent `...Scopes` objects can be keyed to a specific [CAIP-2][] network identifier, or to an entire [CAIP-104][] namespace.
-`scopeObjects` keyed to an entire [CAIP-104][] namespace SHOULD contain a non-empty `references` array to be actionable, making them functionally equivalent to a series of identical `scopeObjects`, each keyed to one of the members of `references` expressed as a [CAIP-2][] scope.
-An empty or absent `references` array SHOULD NOT be interpreted as a namespace-wide authorization (i.e. authorization for ANY network therein), but rather as a null authorization of 0 specified `references`s within that namespace.
-(See [CAIP-217][] for more details on the structure of the typed objects included in these `...Scopes` objects.)
+Each `scopeObject` is keyed by a [CAIP-2][] chain ID.
```

Comment:
This touches on how we don't have a good place to park methods that do not belong to any ecosystem/chain. Right now we're using the `wallet` scope for that, but it is sort of a pain as it isn't a CAIP-2 chain ID.

The other issue is that we don't have a good place to park methods/accounts that belong to an ecosystem but to no specific chain. In the example below, I see `eip155:0` used, but AFAICT the `0` reference is only meant to be used when specifying that an CAIP-10 Account ID is an EOA account. I feel that this is implied both by the wording in the [EIP-155 CAIP-10 spec](https://github.com/ChainAgnostic/namespaces/blob/main/eip155/caip10.md), but also that this specifically is specified on the CAIP-10 spec and not the CAIP-2 spec

Reply:
Good point.  I deliberately only defined the "CAIP-10" usage of chainId 0 at the time because it was only discussed in the context of EOAs (and the older Gnosis Safe convention).  Adding a similar section to the EIP-155 CAIP-2 spec would work if CAIP-2 wanted to be able to use `eip155:0` specifically as a key for `scopeObject`s, for example.  If that's cleaner or more durable long-term than keeping the "wallet:eip155" special-case, let's discuss.